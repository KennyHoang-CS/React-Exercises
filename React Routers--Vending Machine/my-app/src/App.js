import VendingMachine from './VendingMachine';
import './App.css';
import { BrowserRouter, Route } from 'react-router-dom';
import Coke from './Coke';
import Chip from './Chip';
import CupNoodle from './CupNoodle';

function App() {
  return (
    <div>
      <BrowserRouter>
        <Route exact path="/">
          <VendingMachine />
        </Route>
        <Route exact path="/coke">
          <Coke/>
        </Route>
        <Route exact path="/cupnoodles">
          <CupNoodle/>
        </Route>
        <Route exact path="/chips">
          <Chip/>
        </Route>
      </BrowserRouter>
    </div>
  );
}

export default App;
