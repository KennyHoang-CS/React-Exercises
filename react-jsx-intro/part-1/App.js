const App = () => {
    return (
        <div>
            <FirstComponent />
            <NamedComponent name="Kenny" />
        </div>
    )
}

ReactDOM.render(<App />, document.getElementById('root'))