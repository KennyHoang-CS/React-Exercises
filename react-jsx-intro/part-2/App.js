
const App = () => {
    return (
        <div>
            <Tweet
                name="Mike Jones"
                username="Bobby Flay"
                date={new Date().toDateString()}
                message="I love bunnies!"
            />
            <Tweet
                name="Jessica Nguyen"
                username="Angel Fob"
                date={new Date().toDateString()}
                message="Why you do this to me?!?!"
            />
            <Tweet
                name="Steven Delatorre"
                username="Jake Bell"
                date={new Date().toDateString()}
                message="I need to throw fish."
            />
        </div>
    )
}

ReactDOM.render(<App />, document.getElementById('root'))