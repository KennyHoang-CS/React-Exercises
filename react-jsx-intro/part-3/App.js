
const App = () => {
    return (
        <div>
            <Person 
                name="Miko"
                age={ 1 }
                hobbies={ ["eat", "sleep", "play games"]}
            />
            <Person 
                name="Angel"
                age={ 9999 }
                hobbies={ ["Praying"]}
            />
            <Person 
                name="Stephano"
                age={ 56 }
                hobbies={ ["Demon Lord Role Playing", "Eating Souls"] }
            />
        </div>
    )
}

ReactDOM.render(<App />, document.getElementById('root'))