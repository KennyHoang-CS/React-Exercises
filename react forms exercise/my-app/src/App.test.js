import { render, screen } from '@testing-library/react';
import App from './App';


it('App should render', () => {
  render(<App />);
});

it('App render matches snapshot', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot(); 
});