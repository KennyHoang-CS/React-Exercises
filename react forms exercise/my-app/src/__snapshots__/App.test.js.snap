// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div>
      <h3>
        Box List
      </h3>
      <form>
        <label
          for="color"
        >
          Color
        </label>
        <input
          id="color"
          name="color"
          placeholder="box color"
          type="text"
          value=""
        />
        <label
          for="height"
        >
          Height
        </label>
        <input
          id="height"
          name="height"
          placeholder="height"
          type="text"
          value=""
        />
        <label
          for="width"
        >
          Width
        </label>
        <input
          id="width"
          name="width"
          placeholder="width"
          type="text"
          value=""
        />
        <button>
          Add
        </button>
      </form>
      <div />
    </div>
  </div>
</DocumentFragment>
`;
